### PR TITLE
Integrate auth tokens into temporal server connection

### PR DESCRIPTION
**Pull request by feedmeapples**

### CODE REVIEW

File: server/routes.js [hunk 1]
```diff
@@ -14,16 +14,22 @@ const wfClient = new WorkflowClient();
...
-      : undefined,
-  });
+  ctx.body = await wfClient.listNamespaces(
+    {
+      pageSize: 50,
+      nextPageToken: ctx.query.nextPageToken
+        ? Buffer.from(ctx.query.nextPageToken, 'base64')
+        : undefined,
+    },
+    { accessToken: extractAccessToken(ctx) }
```
Comment (sw-yx) [inline server/routes.js:24]: suggestion: the amount of times you are repeating this line of code suggests that there is some kind of middleware that could be used here

File: server/workflow-client.js [hunk 2]
```diff
@@ -293,21 +317,27 @@ WorkflowClient.prototype.openWorkflows = async function({
     typeFilter,
     executionFilter,
   };
-  const res = await this.client.listOpenWorkflowExecutionsAsync(req);
+  const res = await this.client.listOpenWorkflowExecutionsAsync(
+    req,
+    buildGrpcMetadata(metadata)
```
Comment (sw-yx) [inline server/workflow-client.js:322]: suggestion 2: the amount of times you are repeating this line of code suggests that there is some kind of middleware that could be used here

it feels like you are expanding the `openWorkflows` function to receive metadata, only to pass thru metadata to `listOpenWorkflowExecutionsAsync`, and repeating that a few times here. to me a bit of a code smell. i havent looked at this client impl closely but just asking - is there a better way to do what you are trying to do here? are you going to regret adding this in future and have to refactor a bunch?

Comment (feedmeapples) [reply server/workflow-client.js:322]: @sw-yx i fully agree i spent some time yesterday thinking about a better approach and didn't finish as it was a bit late, created this PR for visibility. Is this ok if i refactor straight after we test mtls?

Comment (sw-yx) [reply server/workflow-client.js:322]: sgtm

File: server/routes.js (hunk 1 above)
Comment (feedmeapples) [reply server/routes.js:24]: similar to below,  fully agree this refactoring is needed

Comment (feedmeapples) [reply server/routes.js:24]: just to follow up and visibility:created an item in internal backlog to track this

File: server/workflow-client.js (hunk 2 above)
Comment (feedmeapples) [reply server/workflow-client.js:322]: followin up for visibility:created an item in internal backlog to track this

**Pull request merged**